File: src/main/scala/beam/agentsim/MockModeStatsControlerListener.scala
package org.matsim.analysis

import javax.inject.{Inject, Provider}
import org.matsim.api.core.v01.population.Population
import org.matsim.core.config.groups.{ControlerConfigGroup, PlanCalcScoreConfigGroup}
import org.matsim.core.controler.OutputDirectoryHierarchy
import org.matsim.core.router.TripRouter

class MockModeStatsControlerListener @Inject()(controlerConfigGroup: ControlerConfigGroup,
                                               population1: Population,
                                               controlerIO: OutputDirectoryHierarchy,
                                               scoreConfig: PlanCalcScoreConfigGroup,
                                               tripRouterFactory:Provider[TripRouter])
  extends ModeStatsControlerListener(controlerConfigGroup, population1, controlerIO, scoreConfig, tripRouterFactory) {

  //println("Inside mocked ModeStatsControlerListener")

}

File: src/main/java/beam/agentsim/MockModeStatsControlerListener.java
package beam.agentsim;

import org.matsim.analysis.ModeStatsControlerListener;
import org.matsim.api.core.v01.population.Population;
import org.matsim.core.config.groups.ControlerConfigGroup;
import org.matsim.core.config.groups.PlanCalcScoreConfigGroup;
import org.matsim.core.controler.OutputDirectoryHierarchy;
import org.matsim.core.router.TripRouter;

import javax.inject.Inject;
import javax.inject.Provider;


//public class MockModeStatsControlerListener extends ModeStatsControlerListener {
//
//    @Inject
//    MockModeStatsControlerListener(ControlerConfigGroup controlerConfigGroup, Population population1, OutputDirectoryHierarchy controlerIO,
//                                   PlanCalcScoreConfigGroup scoreConfig, Provider<TripRouter> tripRouterFactory ) {
//        super(controlerConfigGroup,population1,controlerIO,scoreConfig,tripRouterFactory);
//
//    }
//
//}
